motion_tracker/utils/image_generator.py

```diff
@@ -0,0 +1,151 @@
+
+import pandas as pd
+import numpy as np
+import cv2
+from image_cropping import Coords, crop_and_resize
+from crop_vis import show_single_stage 
+
+def master_generator(num_video_crops=10, num_image_crops=10, batch_size=50): 
+    """Generator yielding a batch of images. 
+
+    Args: 
+    ----
+        num_video_crops: int
+            Number of random crops of the current frame to take. 
+            Denoted as k3 in the paper, with a value of 10. 
+        num_image_crops: int 
+            Number of random crops of the randomly sampled image 
+            to take. Dentoed as k4 in the paper, with a value of 10. 
+        batch_size: int
+            Number of items in list to be returned. 
+    """
+    # To switch witch training images we draw from first in 
+    # a batch. 
+    imagenet = True
+    imagenet_gen = imagenet_generator(num_image_crops)
+    alov_gen = alov_generator(num_video_crops)
+
+    while True: 
+        
+        out_imgs = []
+        out_boxes = []
+        video_pairs = 0
+        images = 0
+        num_obs = 0
+        # Might need to add in some check or something in the 
+        # case that the num_video_crops and num_image_crops can't
+        # logically (and in an alternating way) add up to the 
+        # batch size. This while loop figures out how many pairs 
+        # of video frames to grab and how many images to grab. 
+        while num_obs < batch_size: 
+            if imagenet: 
+                images += 1
+                num_obs += num_video_crops
+            else: 
+                video_pairs += 1
+                num_obs += num_image_crops
+            imagenet = not imagenet
+
+        print(images, video_pairs)
+        
+        # Now grab the actual images and video based off the number
+        # of pairs and sets of random croppings we need. 
+        for idx in range(images): 
+            imgs, boxes = next(imagenet_gen)
+            out_imgs.append(imgs)
+            out_boxes.append(boxes)
+
+        '''
+        for idx in range(video_pairs): 
+            imgs, boxes = next(alov_gen)
+            out_imgs.append(imgs)
+            out_boxes.append(boxes)
+        '''
+
+        yield out_imgs, out_boxes
+            
+def imagenet_generator(num_image_crops):
+    """Generator yielding imagenet images and bounding boxes. 
+
+    Args: 
+    ----
+        num_image_crops: int
+            Number of image crops to take of each randomly drawn 
+            image. 
+
+    Outputs:
+    -------
+        imgs_lst: list of images
+        boxes_lst: list of bounding box coords
+    """
+    
+    raw_image_dir = 'data/imagenet/images/'
+    img_metadata = pd.read_csv('work/imagenet/parsed_bb2.csv')
+    output_width = 256
+    output_height = 256
+
+    while True:
+        imgs_lst = []
+        boxes_lst = []
+        img_row = img_metadata.sample(1)
+        
+        # Make sure to read in the image only once, and calculate 
```

Comment:
Awesome.

On Sat, Apr 30, 2016 at 7:58 AM, Sean Sall <notifications@github.com> wrote:

> In motion_tracker/utils/image_generator.py
> <https://github.com/dansbecker/motion-tracking/pull/14#discussion_r61669965>
> :
>
> > +    -------
> > +        imgs_lst: list of images
> > +        boxes_lst: list of bounding box coords
> > +    """
> > +
> > +    raw_image_dir = 'data/imagenet/images/'
> > +    img_metadata = pd.read_csv('work/imagenet/parsed_bb2.csv')
> > +    output_width = 256
> > +    output_height = 256
> > +
> > +    while True:
> > +        imgs_lst = []
> > +        boxes_lst = []
> > +        img_row = img_metadata.sample(1)
> > +
> > +        # Make sure to read in the image only once, and calculate
>
> Sorry, yeah I'll make sure to go out further. For now, those comments were
> somewhat just to highlight what I meant to do there, and I was planning on
> removing some of them after everything was said and done.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/dansbecker/motion-tracking/pull/14/files/ae85f5ab63ad24107fab1fa228800e0fcd08cd06#r61669965>
>